aten/src/ATen/native/Bucketization.cpp

```diff
@@ -89,14 +89,21 @@ void dispatch(Tensor& result, const Tensor& input, const Tensor& boundaries, boo
 
 Tensor& searchsorted_out_cpu(const Tensor& sorted_sequence, const Tensor& self, bool out_int32, bool right, Tensor& result) {
   searchsorted_pre_check(sorted_sequence, self, result, out_int32);
+  Tensor out = result;
   if (result.numel() == 0) {
     result.resize_(self.sizes());
```

Comment:
We should use `at::native::resize_output` for resizing output tensors passed with `out=` parameter. It already checks that either the `out` tensor is empty or it's size is correct with the desired size.